### PR TITLE
fix: remove name requirment from donation

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

This should remove the need for name as a requirement for making stripe donations.

### CODE REVIEW

File: api-server/src/server/utils/donation.js [hunk 1]
```diff
@@ -180,11 +180,12 @@ export async function createStripeCardDonation(req, res, stripe) {
...
       duration
     },
-    user: { name, id: userId, email },
+    user: { id: userId, email },
     user
   } = req;
 
-  if (!tokenId || !amount || !duration || !name || !userId || !email) {
+  if (!tokenId || !amount || !duration || !userId || !email) {
+    console.log();
```
Comment (ShaunSHamilton) [inline api-server/src/server/utils/donation.js:188]:
```suggestion
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I see the field is optional, but I wonder if there is a reason to include it if it exists?

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Comment by hamzawaleed0102** [comment]
Looks good to me!

**Review by hamzawaleed0102 — APPROVED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**